internal/controller/numaflowcontroller/numaflowcontroller_controller.go

```diff
@@ -454,29 +358,54 @@ func determineTargetObjects(
 	if err != nil {
 		return nil, fmt.Errorf("can not parse file data, err: %w", err)
 	}
-	manifestsWithOwnership, err := applyOwnershipToManifests(manifests, controller)
-	if err != nil {
-		return nil, fmt.Errorf("failed to apply ownership reference, %w", err)
-	}
-
-	targetObjs, err := toUnstructuredAndApplyLabel(manifestsWithOwnership, controller.Name)
+	targetObjs, err := toLabeledUnstructured(manifests, controller.Name)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse the manifest, %w", err)
 	}
+	for _, obj := range targetObjs {
+		obj.SetNamespace(namespace)
+	}
```

Comment:
In order to compare the manifests effectively, the target manifests need to have "namespace" applied